Add coc-damage-calculator recipe

S0mbr3 opened the pull request:
### Brief summary of what the package does

coc-damage-calculator package is meant to help Clash of Clans players to quickly calculate damages to buildings from various equipements and spells.

It provides an hydra menu to select the options.

### Direct link to the package repository

https://github.com/S0mbr3/coc-damage-calculator

### Your association with the package

I am the author and maintener.

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Hi @S0mbr3 - please be careful about checking boxes on items that you haven't actually checked re: "I've built and installed the package using the instructions in CONTRIBUTING.org" - if you had checked, you would realize you had an invalid recipe.  I've adjusted the recipe to a working (default) - this default would give your package room to grow within the conventions in the future.

A couple quick notes on the code itself -

- The `;;; Commentary` for this file is much wider than 80 characters
- In fact a lot of the code is unwrapped - one line is 272 characters wide.  A lot of elisp devs (the people you'd expect to look at this code) won't like this, and my recommendation is 80-100 characters per line.
- You are very likely overusing `;;;###autoload` - if anything *at all*, `coc-damage-calculator-menu-hydra` would be the only thing I'd autoload.
- I would rename `coc-damage-calculator-menu-hydra` to `coc-damage-calculator-menu` or even shorter name this `coc-damage` or `coc-damage-hydra` so users have a very quick M-x way to access this interface. 
- Inlining `(require 'hydra)` into the hydra function is probably excessive - just bring in the dependency at the top of the file
- Docstrings like `"Damage FIREBALL-LEVEL EARTHQUAKE-LEVEL EARTHQUAKE-NUMBER ARROW-LEVEL HP."` written solely to satisfy checkdoc are useless - I'd either write the documentation string out properly, or not have a docstring at all.  Checkdoc is within reason on MELPA.

S0mbr3 commented:
Hi @riscy- I've updated a couple things to suit better elisp conventions and melpa quality assurance. 

- I might have not understand how to use autoload properly.
- The docstrings were absolutely useless and I'm very sorry for that.
- inlining the require was obviously overkill.
- Bad habits to rely on code formatters to wrap the code.
- I've changed the package name to have shorter functions names and a quicker access with `M-x`

As you might have guessed fairly easily this is my first package. I might make useful packages in the future. 

Very thanks for your patience and your very helpful insights.

riscy commented:
> I might have not understand how to use autoload properly.

It lets people access/refer to parts of your package before they explicitly load it.  For example python-mode might not be loaded in to your runtime, but you can type `M-x python-mode` at any time, and then python-mode will be autoloaded.   It's probably not necessary in this case (and when in doubt, don't autoload).

> The docstrings were absolutely useless and I'm very sorry for that.

No problem.  :)  My language was probably a little blunt!

> Bad habits to rely on code formatters to wrap the code.

There are a few code formatters for elisp but I think the space isn't very mature, so I sympathize.   :)

Altogether this look good to go and I'll merge.  But I will mention the prompts often write "Press ..." when they should write "Enter ..." - e.g. `Press the level of the rocket spear:` should be written as `Enter the level of the rocket spear:`.  Even more minor, I think "spicky" ball should be "spiky" ball. :)
